packages/nextjs-analytics/src/types.ts

```diff
@@ -0,0 +1,41 @@
+import type { getBuildingBlockSDK } from "@ogcio/building-blocks-sdk";
+
+type AnalyticsClientProps = ReturnType<typeof getBuildingBlockSDK>["analytics"];
+
+interface AnalyticsTrackerProps {
+  /**
+   * The user ID to track.
+   * Ensure that the user ID is unique to the user.
+   * Tracking the UserID should be done in compliance with the GDPR.
+   * @default undefined
+   */
+  userId?: string;
+  customDimensions?: {
+    [x: `dimension${number}`]: string;
+  };
+}
```

Comment:
unless you need to make an override on the comment you can inherit the type 

```suggestion
type AnalyticsTrackerProps = Parameters<AnalyticsClientProps[‘setTrackingContext’]>[0];
```

Reply:
https://github.com/ogcio/shared-node-utils/commit/d0a12a31db975d8376b79c6e6fe960ce6ea26dc6